src/include/argdata.h

```diff
@@ -34,6 +34,12 @@ typedef __argdata_t argdata_t;
 #define _ARGDATA_T_DECLARED
 #endif
 
+typedef struct {
+	__size_t index;
+	int error;
+	__size_t __data[16];
+} argdata_iterator_t;
```

Comment:
I'd rather see that we introduce a separate type for both iterators. These types don't have any overlap in use, both publicly and internally. By using two separate types, there's no way people mix up the API.